### DISCUSSION TITLE
Increase visibility for all `/developers`

### DISCUSSION THREAD

**Discussion by joevans** [joemasilotti/railsdevs.com#407] [Ideas]
This is more of a "do you have any goals in the future for fixing this growing problem?" thread. I count 267 developers on the site currently and developer's visibility is either "recently made themselves available" or "recently created an account." For the middle guys, they won't have any visibility on the "first page."

Proposed temporary/long term solutions:
- Show more devs at once.
- Show devs randomly on every refresh before sorting.
- Have an algorithm around if the users have been viewed on a per user level.
- Gamify the devs. (idk how...but could be worth exploring.)

**Comment by joemasilotti** [comment]
Next big steps are improving filtering via #356. When you say increasing invisibility, what exactly do you mean? Showing a wider variety of folks when someone browses (but doesn't filter) on the site?

**Reply by joevans** [reply]
Oh, thank you for placing it here. I should have done so.

Yes, showing wider variety but regardless if someone filters. I think the same problem currently exists with filtering too.

Maybe when the filters expand, like location, etc, then the problem would change on that end.

**Reply by joemasilotti** [reply]
FWIW, as an admin I have access to the following filters. These will all be exposed to paying customers once #356 is completed. I routinely get a manageable list of developers when applying these.

<img width="1022" alt="image" src="https://user-images.githubusercontent.com/2092156/166303208-ffb0f47e-2be0-433d-b754-d03f30ecfc86.png">

Regardless, if you have ideas on how to improve a specific problem I'm all ears!

**Reply by joevans** [reply]
Understood. I can see those same filters upon fresh installing the app.

I am primarily referring to the fact that, regardless of filters, you always see the same results. IMO, randomizing it a bit would solve the "I've been on the platform for weeks, why can I not see myself in the results?" hypothetical problem.

**Reply by joemasilotti** [reply]
> "I've been on the platform for weeks, why can I not see myself in the results?" hypothetical problem

I'm not sure how much resources I want to dedicate to a hypothetical problem. If this is something you or others are experiencing I'd be happy to work on a potential solution, though!

**Reply by joevans** [reply]
Fair enough.

**Comment by joevans** [comment]
I think this [article you wrote](https://masilotti.com/railsdevs-gender-diversity/) can be tied into this. You could have an randomizing algo that displays other genders up top. This is a unthought through solution but it is a potential solution.

**Reply by joemasilotti** [reply]
I like that idea! It could even rotate the featured developer slot with a random developer or two each week.
